job/server/src/main/java/alluxio/master/job/plan/PlanCoordinator.java

```diff
@@ -197,17 +200,24 @@ public void setJobAsFailed(String errorMessage) {
    */
   public void failTasksForWorker(long workerId) {
     synchronized (mPlanInfo) {
-      Long taskId = mWorkerIdToTaskId.get(workerId);
-      if (taskId == null) {
+      List<Long> taskIds = mWorkerIdToTaskIds.get(workerId);
+      if (taskIds == null) {
         return;
       }
-      TaskInfo taskInfo = mPlanInfo.getTaskInfo(taskId);
-      if (taskInfo.getStatus().isFinished()) {
-        return;
+
+      boolean statusChanged = false;
+      for (Long taskId : taskIds) {
+        TaskInfo taskInfo = mPlanInfo.getTaskInfo(taskId);
+        if (taskInfo.getStatus().isFinished()) {
+          return;
+        }
+        if (!mPlanInfo.getStatus().isFinished()) {
```

Comment:
Does the condition above not guarantee that this is always true if we reach this statement?

Reply:
Nope. The parent plan could have finished because one of its other tasks could have failed.